cdap-ui/app/cdap/components/AbstractWidget/DLPCustomWidget/DLPRow.tsx

```diff
@@ -0,0 +1,449 @@
+  transforms: ITransformProp[];
+  filters: FilterOption[];
+  extraConfig: any;
+}
+
+interface IDLPRowState {
+  fields: string;
+  transform: string;
+  filters: string;
+  transformProperties: Record<string, string>;
+  expanded?: boolean;
+}
+
+type StateKeys = keyof IDLPRowState;
+
+class DLPRow extends AbstractRow<IDLPRowProps, IDLPRowState> {
+  public static defaultProps = {
+    transforms: [],
+    filters: [],
+    extraConfig: '',
+    expanded: false,
+  };
+
+  public state = {
+    fields: '',
+    transform: '',
+    filters: '',
+    transformProperties: {},
+    expanded: false,
+  };
+
+  public componentDidMount() {
+    try {
+      const jsonString = this.props.value;
+      const jsonObj = JSON.parse(jsonString);
+      this.setState({ ...this.state, ...jsonObj });
+    } catch (error) {
+      this.setState({
+        fields: '',
+        transform: '',
+        filters: '',
+        transformProperties: {},
+      });
+    }
+  }
+
+  // Handles onChange for MultiSelect components (filters and fields selectors)
+  private handleChangeMultiSelect = (type: StateKeys, e: string) => {
+    if (type === 'filters') {
+      const noneWasSelected: boolean = this.state.filters.includes('NONE');
+      const noneIsSelected: boolean = e.includes('NONE');
+
+      // Ensuring user can't select 'NONE' along with other filters
+      // TODO: Use another component for the filters selector (https://issues.cask.co/browse/CDAP-16124)
+      if (noneWasSelected && noneIsSelected) {
+        e = e.replace('NONE,', '');
+      } else if (!noneWasSelected && noneIsSelected) {
+        e = 'NONE';
+      }
+    }
+
+    this.handleChange(type, e);
+  };
+
+  // Handles onChange for Select components (transform selector)
+  private handleChangeSelect = (type: StateKeys, e) => {
+    const value = e.target.value;
+    if (value === this.state.transform) {
+      return;
+    }
+    this.handleChange(type, value);
+    if (type === 'transform') {
+      const transformProps = {};
+      this.props.transforms
+        .filter((transform) => transform.name === value)[0]
+        .options.forEach((element) => {
+          transformProps[element.name] = element['widget-attributes'].default || '';
+        });
+      this.handleChange('transformProperties', transformProps);
+      this.handleChange('expanded', Object.keys(transformProps).length > 0);
+    }
+  };
+
+  private handleChangeTransformOptions(values: Record<string, string>) {
+    // Comparing new values to current state
+    const newValuesKeys = Object.keys(values);
+    const oldValuesKeys = Object.keys(this.state.transformProperties);
+    const longerKeysList =
+      newValuesKeys.length >= oldValuesKeys.length ? newValuesKeys : oldValuesKeys;
+    const longerDict =
+      newValuesKeys.length >= oldValuesKeys.length ? values : this.state.transformProperties;
+    const shorterKeysList =
+      newValuesKeys.length < oldValuesKeys.length ? newValuesKeys : oldValuesKeys;
```

Comment:
can you comment on what is the intent here?

Reply:
Added comment